crates/oxc_linter/src/rules/typescript/no_empty_object_type.rs

```diff
@@ -0,0 +1,413 @@
+use oxc_ast::{
+    ast::{TSInterfaceDeclaration, TSTypeLiteral},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::NodeId;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_empty_object_type_diagnostic(span: Span) -> OxcDiagnostic {
+    // See <https://oxc.rs/docs/contribute/linter/adding-rules.html#diagnostics> for details
+    OxcDiagnostic::warn("Do not use the empty object type literal.")
+        .with_help("To avoid confusion around the {} type allowing any non-nullish value, this rule bans usage of the {} type.")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoEmptyObjectType(Box<NoEmptyObjectTypeConfig>);
+
+#[allow(clippy::struct_field_names)]
+#[derive(Debug, Default, Clone)]
+pub struct NoEmptyObjectTypeConfig {
+    /** Whether to allow empty interfaces. */
+    allow_interfaces: AllowInterfaces,
+    /** Whether to allow empty object type literals. */
+    allow_object_types: AllowObjectTypes,
+    /** allow interfaces and object type aliases with the configured name */
+    allow_with_name: String,
```

Comment:
Use `CompactStr` here instead, so we can borrow if possible

Reply:
ah yeah, you have to call `CompactStr::from`. we can leave this for now, it's not a big deal.